src/main/ddrhook1/config-eamuse.c

```diff
@@ -102,16 +101,8 @@ void ddrhook1_config_eamuse_get(
     }
 
     if (!security_id_verify(&config_eamuse->pcbid)) {
-        tmp = security_id_to_str(
-            &DDRHOOK1_CONFIG_EAMUSE_DEFAULT_PCBID_VALUE, false);
-        tmp2 = security_id_to_str(&config_eamuse->pcbid, false);
-        log_warning(
-            "PCBID verification of '%s' failed, fallback to default "
-            "PCBID '%s'",
-            tmp2,
-            tmp);
-        free(tmp);
-        free(tmp2);
+        log_fatal("PCBID verification failed");
```

Comment:
Suggestion: It might make sense to output the data that failed verification here. Same for the EAMID below.

Reply:
I've actually got some interesting feedback on this whole thing - the PCBID that is provided in config is totally different from the PCBID any server gets (bytes are shuffled about, checksum is removed, prefix is forced), and it's extremely confusing why on earth this happens to the server administrator.
I think it'd be better to just force 0140 as the prefix (which is what the game does), auto-calculate the checksum, and send that instead (because the checksum doesn't make it to the server at all).

This isn't just a server thing, but the test menu's SYSTEM page displays the same "broken" PCBID.

Reply:
Huh, that's a problem I ran into also with some of the old IIDX games. You brought this up in a chat already and the suggestion was to handle the unscrambling etc. on the hook side, so the game gets a correctly ordered and non broken ID.

Since handling of PCBID/EAMID is already known to be different across different games and also versions of the same game series, I agree to the approach that we also stick to what jubeat does and implement that specifically and only for the jubeat hooks.

To summarize, that means:

* Allow users to set any clean PCB/EAM ID in the hook config. Stick to the, what I believe is, "standard base format", e.g. `0101020304050607083F`: first `01` being the header, `3F` the checksum and `01020304060708` being the actual ID.
* The hook takes care of any game specific needs, in this case for jbhook:
  * Shuffle bytes so the server gets the ID in exactly the same way as the user provided it in the hook config
  * Enforce `0140` as prefix and throw error to user that PCBIDs need that prefix
  * Checksum verification and tell the user if mismatch
  
Does that make sense? Anything missing?

Reply:
I think the user shouldn't need to provide the checksum - with the `0140` prefix, the checksum is invisible as far as the server is concerned, with a remaining 8 bytes (16 characters stringified) available for the "actual" PCBID.

To be clear, the PCBID:
`01020304050607080926`
Will be displayed in the test menu, and sent to any server as:
`01400208070605040309`

Reply:
Hmm, that would differ from how other games supported in bemanitools require those IDs to be provided by the user as well as the "official way" in an `ea3-config.xml` on newer games. I expect that telling the user to strip the existing IDs to be quite error prone.

Reply:
I think we could argue that we've been doing it wrong this whole time - I've now talked to 3 server operators who essentially said "PCBIDs on old games get mangled when they're sent to us".

I think this could be adapted to be different logic - 
- If the checksum is valid, assume the user has an old config, and use the old behaviour
- Otherwise, enforce new "matches the test menu" behaviour (aka `0140` prefix)

Reply:
Can you elaborate which games and versions specifically are known to be affected by this?

Changing the logic on other games, e.g. IIDX, to find common alignment is a change with a larger blast radius. I suggest to push that back for now until we understand how many different ways we have to deal with and which games and versions are using them.

Reply:
Yeah look, that's fair. It's going to be a breaking change, considering it's been acceptably broken for so long.

I can revert the security key stuff and postpone that to a separate commit instead of bundling it with jubeat.

As far as I can tell, it affects p3io games (DDRX, jubeat), ezusb games (IIDX...?). I almost want to generalise and say "all roundplug games" but it might be closer to "all roundplug2 games" since I noticed a few different versions of round plugs in the source.

Reply:
Ok, cool. I like the approach to de-couple this from the other changes in this PR that I am happy to merge. I suggest to also lift this to an issue where we can continue discussing about how to address this on other games. One of the key problems I see is that we also need a solution, potentially a migration path, for server operators supporting already the "broken way" of doing this. I expect we have to keep supporting this for a long time, probably forever on the games where we are supporting it already.